### PR TITLE
Show tags

### PR DESCRIPTION
**Pull request by GretaD**
Fixes https://github.com/nextcloud/mail/issues/4805

- [x] Show the labels if theres labels added by another mail client
- [x] Show the right color to each tag
- [x] Fix the styling
- [x] Use ListItem component instead of divs
- [x] Improve styling on small screen. Its okey-ish how its now but can be better

### CODE REVIEW

**Comment by ChristophWurst** [comment]
you can rebase now :)

File: src/components/Envelope.vue [hunk 1]
```diff
@@ -314,9 +325,25 @@ export default {
...
+	}
 
 	.app-content-list-item-details {
 		padding-right: 7px;
 	}
 }
+.tagLabel {
+	z-index: 1;
+	position: relative;
+	border: 1px solid #FFC300;
```
Comment (GretaD) [inline src/components/Envelope.vue:339]: not the right styling, i did it just to see where my tags are, will fix it

**Comment by GretaD** [comment]
taglabel

**Comment by ChristophWurst** [comment]
> The state right now

@nimisha-vijay :eyes: :)

**Comment by GretaD** [comment]
look at this beauty

taggin

**Review by ma12-co — COMMENTED** [review]
Nice one :) 
Just one question, don't you think that those tags are a bit to prominent? Both in terms of size and vibrancy of the colors? 
cc @jancborchardt and @nimisha-vijay

**Comment by nimisha-vijay** [comment]
This looks so cool! :D I agree that the colours are a tiny bit too intense. Maybe we can do something similar to Deck?
image

Or have it with a full opacity border but lower opacity fill? (dunno if this is used in any other apps)
image

What do you guys think? @ma12-co @ChristophWurst @jancborchardt

**Comment by ma12-co** [comment]
I like deck solution but I also like that in deck the font-size of the tag is smaller than the title. Looking at Greta's screenshot, the tags are much more prominent than sender's name and subject, which is not good. So on top of making colors lighter, I'd shrink them a bit too.

**Comment by nimisha-vijay** [comment]
Yes correct, decreasing text size sounds good. Maybe also decreasing the padding in the tag by a couple of pixels might make it appear less prominent. 
And maybe align the left of the tags list with the text? So that there is nothing vertically in between the avatars. Does that make sense?

**Comment by ma12-co** [comment]
> Yes correct, decreasing text size sounds good. Maybe also decreasing the padding in the tag by a couple of pixels might make it appear less prominent.

Agree with reducing padding too!

> And maybe align the left of the tags list with the text? So that there is nothing vertically in between the avatars. Does that make sense?

Not sure about this one, might create a weird empty spot to the right, I'd have to see that in the a list to see how that would work..

File: src/components/Envelope.vue [hunk 2]
```diff
@@ -223,100 +343,133 @@ export default {
...
+	padding-right: 7px;
+	}
+	}
+	.tagLabel {
+	margin: 0 10px;
+	}
+	.tagGroup {
+	display: inline-block;
+	border: 1px solid transparent;
+	border-radius: 35px;
```
Comment (ma12-co) [inline src/components/Envelope.vue:448]:
```suggestion
	border-radius: var(--border-radius-pill);
```

File: src/components/Envelope.vue [hunk 3]
```diff
@@ -223,100 +343,133 @@ export default {
...
 
 	.app-content-list-item-details {
-		padding-right: 7px;
+	padding-right: 7px;
+	}
+	}
+	.tagLabel {
+	margin: 0 10px;
+	}
+	.tagGroup {
```
Comment (ma12-co) [inline src/components/Envelope.vue:445]:
```suggestion
	.tagGroup {
	font-size: calc(var(--default-font-size) * 0.8)
```

**Review by ma12-co — CHANGES_REQUESTED** [review]

**Comment by GretaD** [comment]
> Or have it with a full opacity border but lower opacity fill? (dunno if this is used in any other apps)
> image
> 


@nimisha-vijay what are the codes for these colour? background and border. Jan agreed to have this background, but not border. But i need the border colour to add it to the text.

**Comment by nimisha-vijay** [comment]
image

Important: #FF7A66
Todo: #317CCC
Finished: #31CC7C
Personal: #A85BF7
All at 100% opacity for text, 15% for background. 
Later: #DFCC54 at 15% opacity for bg, and #7E6D00 at 100% opacity for text

cc @jancborchardt looks good?

**Comment by ChristophWurst** [comment]
> Important: #FF7A66
> Todo: #317CCC
> Finished: #31CC7C
> Personal: #A85BF7
> All at 100% opacity for text, 15% for background.
> Later: #DFCC54 at 15% opacity for bg, and #7E6D00 at 100% opacity for text

We can work with some wisely picked defaults for the five *default tags*. But after #4806 the users will be able to assign *any* color to their tags. Will the opacity for text/background work with that?

**Comment by nimisha-vijay** [comment]
As long as there is enough contrast between text and bg, it should be fine. Lighter colours will probably not work because the text would not be visible (like the yellow tag, later, used a different text and bg colour because otherwise it would not be visible). 
If we are using a colour picker palette, maybe we can explicitly specify values for text and bg colour for each colour in the palette to make sure there are no visiblity/contrast issues?

**Comment by ChristophWurst** [comment]
> [ @GretaD ](/GretaD)
> ` [fixup! Merge branch 'master' into enhancement/show_tag](/nextcloud/mail/pull/4894/commits/d6e5e5798948a61f6edd23d86da41248ad4d3c1f)`

:warning: is this really a fixup of the *merge*? Please double check. The rebase might drop your changes otherwise.

File: package.json [hunk 4]
```diff
@@ -39,7 +39,7 @@
     "@nextcloud/logger": "^1.1.2",
     "@nextcloud/moment": "^1.1.1",
     "@nextcloud/router": "^1.2.0",
-    "@nextcloud/vue": "^3.9.0",
+    "@nextcloud/vue": "^4.0.0-alpha.3",
```
Comment (ChristophWurst) [inline package.json:42]: v4 will pull in `@nextcloud/router` v2 and that will drop some of the browsers that we still support or which are still supported for Nextcloud 20-21

**Review by ChristophWurst — CHANGES_REQUESTED** [review]
Making sure we don't merge this lightly

File: package.json (hunk 4 above)
Comment (ChristophWurst) [reply package.json:42]: We decided to drop older browsers.

**Comment by ChristophWurst** [comment]
- [ ] BUG color indicator should extend to the full height

Bildschirmfoto von 2021-05-19 13-36-51

**Comment by ChristophWurst** [comment]
- [ ] BUG important messages have a large invisible area at which they become clickable. If there is a message blow, you can't select that one

Bildschirmfoto vom 2021-05-19 13-37-47

**Review by ChristophWurst — APPROVED** [review]
Works :v:

**Pull request merged**